### ISSUE TITLE
showing/hiding a layer in figma causes replacement context to fail 

### ISSUE THREAD

**Issue by HooHaa-code** [google/automotive-design-compose#1056]
I've created a component with two variants and an icon.  I am replacing the component on the stage with the variant component, and replacing the icon within the variant with the desired icon.  All of the replacements work as intended, until I unhide a second instance of the component on the stage.  When i do that, the working icon is always rendered in which ever state the second icon is set two in figma.  Below is a video of the issue:

https://github.com/google/automotive-design-compose/assets/64603080/b47511b1-29d8-477f-bb03-300b98f47292

Here is a zip file with the source code and figma file
[Ex4Google-variant-icon.zip](https://github.com/google/automotive-design-compose/files/15179663/Ex4Google-variant-icon.zip)

**Comment by timothyfroehlich** [comment]
@HooHaa-code Putting aside the bug that you've reproduced (I see it on my end too), would you mind clarifying what exactly you're trying to do? Like, coding aside, what are you trying to get the running app to do?

**Comment by HooHaa-code** [comment]
I want to have a button with multiple states, with an icon that I can set though code.  So the Icon button would have whatever states the button has, with a replacement context to control which icon is shown

**Comment by timothyfroehlich** [comment]
Just one instance of the button or two?

**Comment by HooHaa-code** [comment]
In the case I need to implement probably 3-4.  The idea is that the button states all look the same but they control different aspects of the app.  So I can set one button style and just swap the icon to indicate which feature of app i want to control.

**Comment by yiqunw700** [comment]
@HooHaa-code I am not sure if there is something off in the figma design file. I tried deleting the icon2 and insert it again without altering the icons and the issue is gone.
![image](https://github.com/google/automotive-design-compose/assets/165221276/cddc9a1c-cac7-425d-81d6-522505435385)

I will also investigating further to understand why icon2 updates icon1 in your report.

**Comment by yiqunw700** [comment]
It also works when icon2 looks like this:
![image](https://github.com/google/automotive-design-compose/assets/165221276/536ab5d8-dd0c-4ce2-9363-d48318a89edb)
But does not work when icon2 has #iconContainer replaced
![image](https://github.com/google/automotive-design-compose/assets/165221276/f455de86-68a5-41aa-a7d3-694512a283e4)

**Comment by HooHaa-code** [comment]
I will attempt to describe what I would like to accomplish though user stories:

- As a developer, I want a composable to set the state and icon of a button:
```
enum class State {
    Disabled,
    Enabled,
    Selected
}

@DesignComponent(node = "myButton")
    fun myButton(
        @DesignVariant(property = "state") state: State,
        @Design(node = "#icon") icon: @Composable (ComponentReplacementContext) -> Unit,
    )

@Composable
    fun Example (){
        ExampleDoc.mainFrame(
            button1 = { myButton(Icons.HEADPHONES, State.Enabled) },
            button2 = { myButton(Icons.HEAD_SIDE_THINKING, State.Disabled) },
            button3 = { myButton(Icons.HEADPHONES, State.Selected) },
            button4 = { myButton(Icons.HEADPHONES, State.Selected) },
        )
    }
```

- As a designer I want to create all of my button states without an icon.  After the states are created, I would like to add any icon to that button:
![image](https://github.com/google/automotive-design-compose/assets/64603080/26ecf108-974c-4bdd-9005-083892823eb1)

- Possible use case:
I am creating an app that controls my dishwasher.  The dishwasher has many modes that are supported: quick, normal, heavy duty, overnight, etc.  Each mode is represented by a button, and in that button an icon to indicate the mode.  If the cycle is not running, I would like the buttons to be pressed by the user to indicate which modes they would like to run.  If the cycle is running, I would like to buttons to indicate which mode is currently running.

What would the code and figma file need to be designed to achieve this.

Due to NDA restrictions I cannot share the actual use case I am attempting to implement via github.  I will be willing to share designs during our next meeting that is covered by our NDA

**Comment by yiqunw700** [comment]
After debugging further:

1. Figma returns the node list {0}: Background, {1}: #Icon2 with child IconButton@2:29 {2}: #Icon1 with child IconButton@29:37
2. SerializedDesignDoc saved both #Icon2 and #Icon1 with child pointing to IconButton@2:29 
3. Because Figma has made a change to #Icon2 that the IconButton@2:29 doesn't have a #iconContainer as I commented earlier, the customization in the code doesn't replace the component with the given icon. 

> @DesignComponent(node = "IconButton")
> fun iconButton(  
>      @DesignVariant(property = "state") state: IconButtonState,
>     @Design(node = "#iconContainer") icon: @Composable (ComponentReplacementContext) -> Unit,
> )

**Comment by rylin8** [comment]
@HooHaa-code FYI you can temporarily get around this by having the property name of `IconButton` be the same name -- `IconButton` instead of `state`. This has been a limitation we've had for quite awhile but hasn't been documented. We'll have a fix for this likely in 0.27, but you can do this to get around the issue for now.
